deployment/terraform/batch-container-service.tf

```diff
@@ -0,0 +1,95 @@
+#
+# Security group resources
+#
+resource "aws_security_group" "batch_container_instance" {
+  vpc_id = "${module.vpc.id}"
+
+  tags {
+    Name        = "sgBatchContainerInstance"
+    Project     = "${var.project}"
+    Environment = "${var.environment}"
+  }
+}
+
+#
+# Autoscaling Resources
+#
+data "template_file" "batch_container_instance_cloud_config" {
+  template = "${file("cloud-config/ecs-batch-user-data.yml")}"
+
+  vars {
+    ecs_cluster_name = "${aws_ecs_cluster.batch_container_instance.name}"
+    environment      = "${var.environment}"
+  }
+}
+
+resource "aws_launch_configuration" "batch_container_instance" {
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  ebs_optimized = true
+  # TODO: Use a separate container instance profile for Batch cluster?
+  iam_instance_profile = "${aws_iam_instance_profile.container_instance.name}"
+  image_id             = "${var.ecs_instance_ami_id}"
+  instance_type        = "${var.batch_container_instance_type}"
+  key_name             = "${var.aws_key_name}"
+  # TODO: Use a separate security group for Batch cluster instances?
+  security_groups      = ["${aws_security_group.container_instance.id}"]
+
+  user_data = "${data.template_file.batch_container_instance_cloud_config.rendered}"
+
+  # Not necessary for NVMe SSD instance stores
+  # ephemeral_block_device {
+  #   device_name = "${var.batch_container_instance_ebs_mount_point}"
+  #   virtual_name = "ephemeral0"
+  # }
+}
+
+resource "aws_autoscaling_group" "batch_container_instance" {
+  name = "asg${var.environment}BatchContainerInstance"
+
+  launch_configuration      = "${aws_launch_configuration.batch_container_instance.name}"
+  health_check_type         = "EC2"
+  desired_capacity          = "${var.batch_container_instance_asg_desired_capacity}"
+  min_size                  = "${var.batch_container_instance_asg_min_size}"
+  max_size                  = "${var.batch_container_instance_asg_max_size}"
+
+  enabled_metrics = [
+    "GroupMinSize",
+    "GroupMaxSize",
+    "GroupDesiredCapacity",
+    "GroupInServiceInstances",
+    "GroupPendingInstances",
+    "GroupStandbyInstances",
+    "GroupTerminatingInstances",
+    "GroupTotalInstances",
+  ]
+
+  vpc_zone_identifier = ["${module.vpc.private_subnet_ids}"]
+
+  tag {
+    key                 = "Name"
+    value               = "BatchContainerInstance"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "Project"
+    value               = "${var.project}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "Environment"
+    value               = "${var.environment}"
+    propagate_at_launch = true
+  }
+}
+
+#
+# ECS Resources
+#
+resource "aws_ecs_cluster" "batch_container_instance" {
```

Comment:
Ah, the connection is made via the `cloud-config` YAML: https://github.com/azavea/pfb-network-connectivity/blob/feature/awf/aws-batch-unmanaged/deployment/terraform/cloud-config/ecs-batch-user-data.yml#L21-L25

You just need the name passed into that template as `ecs_cluster_name`.